pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/JsonMailboxIdentifier.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.mailbox;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Objects;
+
+
+public class JsonMailboxIdentifier implements MailboxIdentifier {
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private final String _jobId;
+  private final String _from;
+  private final String _to;
+
+  private final ServerAddress _fromAddress;
+  private final ServerAddress _toAddress;
+
+  @JsonCreator
+  public JsonMailboxIdentifier(
+      @JsonProperty(value = "jobId") String jobId,
```

Comment:
IIUC jobID is also a combo string of requestID + stageID 
is it possible to also change this to a job object similar to serverAddress? 
later on with support of LogicalSpool we might actually need to expand the support to requestID + senderStageID + receiverStageID

Reply:
yeah, I think for jobId it might make most sense to actually make that a full blown JSON object WDYT?

Reply:
talked offline, if we want to add more we can add them as top level concepts or (if they don't need to be deserialized) we can just encode them into the `jobId` string

Reply:
yes discussed offline and here is the summaryL:
- making it a full blown JSON next will probably cause another backward compatibility issue but will make it much more flexible 
- refactoring it as a jobID object with variable member field will not cause backward compatibility issues but it doesn't need to stack on this PR.